### PR TITLE
LOG-3076: Report to New Relic per Hopper service.

### PR DESCRIPTION
**Pull request by mjkillough**
If `BASE_NEW_RELIC_APP_NAME` is set, report to New Relic for each Hopper
service individually *and* report everything to one central New Relic
application. If `NEW_RELIC_APP_NAME` is specified, report to one New
Relic application as before.

The Reliability Monitor requires that each web/worker service reports to
New Relic separately in order to get statistics about request
counts/errors.

### CODE REVIEW

File: lib/roo_on_rails/railties/new_relic.rb [hunk 1]
```diff
@@ -12,6 +12,15 @@ class NewRelic < Rails::Railtie
 
           abort 'Aborting: NEW_RELIC_LICENSE_KEY is required' if license_key.nil?
 
+          # Report application stats to a per-service (worker, web) New Relic app, and to a main
+          # application for all services.
+          base_name = ENV['BASE_NEW_RELIC_APP_NAME']
+          service_name = ENV['HOPPER_SERVICE_NAME']
+          if !base_name.nil? && ENV['NEW_RELIC_APP_NAME'].blank?
```
Comment (me) [inline lib/roo_on_rails/railties/new_relic.rb]: Should these `nil?` checks be `blank?`?

File: lib/roo_on_rails/railties/new_relic.rb [hunk 2]
```diff
@@ -12,6 +12,15 @@ class NewRelic < Rails::Railtie
...
+            task_app_name = !service_name.nil? ? "#{base_name} - #{service_name}" : nil
```
Comment (me) [inline lib/roo_on_rails/railties/new_relic.rb]: Might be more readable with `service_name.present?`

File: lib/roo_on_rails/railties/new_relic.rb (hunk 1 above)
Comment (mjkillough) [reply lib/roo_on_rails/railties/new_relic.rb]: I think you're right.

File: lib/roo_on_rails/railties/new_relic.rb (hunk 2 above)
Comment (mjkillough) [reply lib/roo_on_rails/railties/new_relic.rb]: Ahh, yes! :-) Much nicer.

**Pull request merged**